### ISSUE TITLE
Some websites are slow and freeze the browser

### ISSUE THREAD

**Issue by BerserkSoul1989** [Feodor2/Mypal68#502]
Hi there, Youtube and Reddit lag and are slow to charge on my PC, mostly Reddit.

I have Windows 7 SP1 and I'm using latest MyPal release

**Comment by bluedragon-cairo** [comment]
> why do you take mypal for win7

I actually use it too because Firefox 70+ removes many traditional and classic about:config settings and it's so annoying, and even newer versions (90+) changes form control skin to some ugly one instead of OS's native

**Comment by Feodor2** [comment]
@bluedragon-cairo 
Do you use yotube or reddid? I do not, but I check yt often and always at the new version, I did not observe a freeze.
I only can tell that latest versions performance drop and more memory hungry, shell sort out when javascript update will be complete.

**Comment by bluedragon-cairo** [comment]
In XP, slightly freezes but not that inconvenient. Also I currently use old.reddit.com (Old reddit site) and Project VORAPIS V3 (a program that replaces the YouTube frontend to 2013 version) so it's much faster because there are no javascript abomination garbages and also it's more convenient to use with nicer look (at least for me)

**Comment by Yggdrax** [comment]
I'd recommend using an invidious/libreddit instance if you are having issues with those websites in the meantime, also @Feodor2, I don't think the issue is that they freeze, but rather that they are too slow to load and to navigate, I've also experimented performance issues with both websites, but mostly with reddit.

**Comment by Feodor2** [comment]
So if are they slow because of their nature, invdius experiences bad times today and going to disable medium format by default that, with its broken format switch, makes it unusable for me, I shall try  VORAPIS / v3.

**Comment by bluedragon-cairo** [comment]
> VORAPIS / v3.

https://addons.mozilla.org/en-US/firefox/addon/v3-get-old-youtube-layout/

----

For Pale Moon/New Moon/Mypal 28-29:
Install Greasemonkey for Pale Moon fork and https://vorapis.pages.dev/product/v3/game_service/patcher/get_latest_client_demo

**Comment by Rekrullurker** [comment]
Up until today, YouTube has been working OK for me. Today it's borderline unusable. Video pages take forever to load, and even after the page seems to be fully loaded, there's still a delay before I can start the video. Trying to use the seek bar is hopeless. Not only does it not show me a thumbnail of that position, like it was doing yesterday, but half the time it won't even move.

Switching between the normal player and fullscreen takes a good 30-45 seconds. If I scroll down, it loads the comments one at a time.

Even on the main page, when I mouse other a thumbnail, it takes a good 20-30 seconds for the pointer to change to the hand image, and then when I move off the thumbnail, it stays in the hand image, even when moving it over parts of the page that aren't clickable. Needless to say that the video previews you're supposed to get when you hover on a video aren't working. Or maybe they do, I lose patience after waiting a couple minutes.

It's like my system is bogged down with something, but only MyPal68 is affected, everything else is working fine.

My old version of Opera doesn't have any extra slowdowns, but then again, video playback is literally the only thing that works. I'd use it for watching videos if the ad skip button worked.

MyPal68 was my last, best option for a fully functional YouTube, and now for whatever reason, it's become a colossal pain in the *ss to use.

I've tried rebooting, I've tried running a clean copy of the latest version, nothing makes any difference.

**Comment by bluedragon-cairo** [comment]
> Today it's borderline unusable. Video pages take forever to load

What's your CPU and RAM? For me the official frontend works not so bad in Pale Moon which is normally slower than Mypal68.

Or try using the V3 frontend, in my opinion it's much faster and easier to use. If you have used YouTube since 2013 or earlier it would be friendly. If you have repetitive playback errors after about 10 seconds of playing let me know because there is a WORKARound.

**Comment by Feodor2** [comment]
I just looked ytube and did not notice changes, Its slow as always but usable good.

**Comment by JimMP68User** [comment]
I also notice slow-down in some YT action - click on notification bell (upper right) and it takes tens of seconds for the drop-down notification list to show up. This started a few days ago. I think it is YT behavior changes.

It might be related to my using the ad-blocker uBlock Origin maybe? 

Also behavior of YT Shorts has gotten sluggish, and it takes tens of seconds for click on Like button to take effect.

Computer Dell 755 4GB memory, Win Xp SP3, Mypal 14.4b and using uBlock Origin adblocker.

All features on YT work, but, delays now when clicking on something.

**Comment by Rekrullurker** [comment]
YouTube seems to be working much better now.

Confession: I've been using version 12.5b for a while now. I tend to not update software if what I have is working and the new version doesn't have any new features I want. When I posted my first message, this version was getting multiple "Unresponsive script" warnings on every video page. 

That's when I downloaded the latest version and tried it. NO script warnings, but everything was taking forever. 

Also, I occasionally had an odd problem where I would click on one thing and it would think I clicked somewhere else. I clicked a video on the front page and it opened a different video instead. The video it opened, was above the one I clicked on. Then, i clicked on the speaker control to mute the ad during an ad, and it opened the author's eshop page.

Just for the hell of it, I loaded version 12.5b tonight and it seems to be working normally. No script warnings, everything responds in a reasonable amount of time, the seek bar works, etc.

I guess it was a temporary problem with YouTube.

**Comment by bluedragon-cairo** [comment]
YouTube developers all just have loose screws these days, they don't know how to optimize JavaScript, they don't know when to use JavaScript and just overuse it everywhere, they remove useful features like playlist translations, they hide dislike counts, they only know chrome, they change site UI and decorations every time, much more

**Comment by Feodor2** [comment]
Tried the VORAPIS, it was installed, but the youtube itself do not load at all after this, then I removed the addon and it is fine again.
So something is broken, but I shall bother not, no time.

**Comment by Rekrullurker** [comment]
YouTube has been working fine for me. Well...

Sometimes videos, especially shorts, will get stuck in one spot for a long time. Sometimes the video will freeze while the audio continues. If I open too many videos in tabs, my system will start thrashing my C drive, and then some of the tabs will be blank when I click on them, and then it will take a very long time and a lot of disk thrashing before the page loads. 

In other words, perfectly normal behavior for YouTube.

**Comment by Feodor2** [comment]
Today I observed glitches on pentiom dual 6500, then I have changed format to avc (h264fy) - no glitches.

**Comment by mike9472** [comment]
Youtube is basically unwatchable for me, keeps freezing up.  Works fine when I switch to Window 10 and the latest Firefox (I have a dual boot computer).  When I'm on Windows XP and Mypal 68.14.4b, it's a no go.  So far it's the only website that gives me problems.

(AMD Quad-Core 3.8GHz / 1TB SSD / Max Ram / 2GB DDR5 Video Card)

**Comment by Rekrullurker** [comment]
I heard something about improper audio/video muxing causing playback problems, and that recent versions of Firefox have supposedly fixed this issue.

I'm seeing three separate problems;

1. Sometimes the video freezes while the audio continues playing,then after 5-10 seconds, the video jumps ahead to catch up to the audio. If I go back in the video, it will freeze at exactly the same spot every time. However if I switch to a different resolution, that section will then play in the new resolution.

2. Sometimes the video will just freeze, I'll get the swirling buffering indicator for 20-30 seconds, then the image will go black for 1-2 seconds, then it will come back and the video will continue to play. Switching to a different resolution will cause the video to almost immediately resume in the new resolution. Also, if I press the right arrow to skip ahead in the video, this will usually cause it to resume playing, at the cost of missing a few seconds. Although sometimes I have to press it 2-3 times before I've gone far enough to get it to resume. 

On some occasions, if I don't intervene, it will just stay on the swirling indicator indefinitely. I've left videos playing (muted) in the background in order to add to the view count for the author's benefit, and a few times I've come back an hour later to find it stuck at 10 minutes in. 

3. Sometimes the video will just skip ahead 1-5 seconds for no apparent reason. If I go back, that section of the video usually plays normally.


If I didn't know any better, I'd think YouTube was intentionally trying to make watching videos as sh*tty as possible in any web browser that isn't Chrome.

**Comment by Rekrullurker** [comment]
I should also mention that the more YouTube videos I view, the slower and more sluggish Mypal68 gets, until I have to close and re-open it.

I go to YouTube's front page, and middle click 3-4 videos to open them in tabs. I have auto-play set to disabled, so it won't load the video until I click the player. After watching each video, I close the tab. 

After I've watched a bunch of videos, YouTube becomes very sluggish. Switching between windowed and fullscreen can take 10-20 seconds. Putting the mouse on the seek bar can take up to 20 seconds before a preview image appears, and then moving the mouse causes another 20 second wait before the preview image updates. Clicking on that position to jump to that point in the video also takes at least 10-20 seconds, and that's BEFORE it moves the position indicator and starts buffering at the new position. If I try to enter a comment, 5-6 letters will appear and then it will lag for several seconds, then the rest of my typing will show up. It goes off into limbo while I'm just moving the cursor back to correct a typo. Even just clicking to pause the video often doesn't register for 15-20 seconds.

After I close Mypal68 and re-open it, everything is back to normal, and YouTube is fairly responsive. Until I view too many videos and the same thing happens again.

I installed an extension that keeps tabs from loading until they're selected, but this didn't make any difference.

**Comment by zanud** [comment]
@Rekrullurker

My experience with Youtube is the same exactly.

**Comment by Feodor2** [comment]
webm codec playing will be fixed in the new version, plenned by the end of the year.
until this use h264fy.

Again I check on the pentium 6500 (2.93x2, 2ram, 8600gt) quite usable and plays fine

**Comment by Rekrullurker** [comment]
I don't understand. How do I tell YouTube to use a specific codec or stream format?

**Comment by zanud** [comment]
> How do I tell YouTube to use a specific codec or stream format?

Add-on [h264fy](https://addons.mozilla.org/ru/firefox/addon/h264ify/) does this automatically.

**Comment by Rekrullurker** [comment]
I installed that add-on and nothing has changed.

Sometimes the video pauses while the audio continues.

Sometimes everything freezes and I get the swirling buffering symbol for 15-20 seconds, the screen goes black for 2-3 seconds, then the video resumes.

Sometimes it just randomly skips ahead 3-5 seconds.

To rule out routing problems between me and the video server, I used the command line program yt-dlp to download the same videos in the same resolution. They downloaded quickly, with no pauses in the data transfer.

I should note that not every video does this. Some channels seem immune while others seem to be swamped with these problems.

**Comment by zanud** [comment]
> I should note that not every video does this. Some channels seem immune while others seem to be swamped with these problems.

Apart from VP8/VP9 Youtube uses some other videocodecs which h264ify can not handle (unlike its more advanced derivatives).

Press RMB on a problematic video, select "Stats for nerds" menu item, and look for a videocodec name in the "Codecs" line - is it "avc1" or not?

**Comment by Rekrullurker** [comment]
OK, I checked the codec on every video I watched tonight and with one exception, they were all vp09. The codec lines looked like this;

vp09.00.51.08.01.01.01.01.00 (244) / opus (251)

The first number in parenthesis was always 244 or 247, and the second was always 250 or 251.

I could watch a video from one channel and not have a single problem, then watch a video from another channel and it would be pausing every 20-30 seconds, but the codec lines in both videos would be exactly the same.

I compared several videos (pausing/not pausing), and often the codec lines were identical. Maybe sometimes, one of the numbers in parenthesis would change, as noted above, but many times they were exactly the same across two different videos, one with problems and one without.

There doesn't seem to be any pattern at all to it. Some channels just seem to be cursed.

**Comment by zanud** [comment]
> I checked the codec on every video I watched tonight and with one exception, they were all vp09

You saw this, despite add-on is installed and enabled?

> I could watch a video from one channel and not have a single problem, then watch a video from another channel and it would be pausing every 20-30 seconds, but the codec lines in both videos would be exactly the same.

Type of codec, by itself, is not a problem. In the past I was involved into video editing (more precisely - into fixing of video, including binary editing of streams). Quite often, streams contains incorrect data, or their structures violate specifications.

Some decoders are quite tolerant to these defects and display such video almost flawlessly. But some are picky and stop decoding till the begin of the next - healthy - fragment of the stream. This is almost exactly what we see currently.

So, perhaps @Feodor2's statement:
> webm codec playing will be fixed in the new version

means: "Webm decoder will be chanded to be more tolerant to different defects".

**Comment by Rekrullurker** [comment]
Apparently the add-in wasn't installed. I have no idea what happened. I installed it the other day and it was listed, but when I checked tonight, it wasn't there. I installed it again, and now it's working. No more random pauses.
